sdk/Trace/Tracer.php

```diff
@@ -124,6 +124,10 @@ public function setActiveSpan(API\Span $span): void
         $this->tail[] = $this->active;
 
         $this->active = $span;
+
+        // FIXME: This should either be called manually or as part of a dedicated tracer operation,
+        // probably some sort of `inSpan` method?
+        $span->makeCurrent();
```

Comment:
This relates to the 2nd major thing that needs to be done. Essentially this should either be called manually by the user, or we have an `inSpan` method that makes a yielded span active for the duration of a callback. Otherwise it might be hard to keep a dedicated method that creates a span and makes it the current one since it would need to return _both_ the span itself and a `Scope` to reset the active span.

Reply:
The Java pattern seems reasonable to me.  Leaving those details to the user might make things more flexible in the long run.  We can always update it later if we get complaints / don't love the implementation details.